pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -118,20 +116,19 @@ def _validate_condition_metric(self, logs):
     def monitor_op(self):
         return self.mode_dict[self.mode]
 
-    def state_dict(self):
+    def on_save_checkpoint(self, trainer, pl_module):
         return {
             'wait_count': self.wait_count,
             'stopped_epoch': self.stopped_epoch,
             'best_score': self.best_score,
             'patience': self.patience
         }
 
-    def load_state_dict(self, state_dict):
-        state_dict = deepcopy(state_dict)
-        self.wait_count = state_dict['wait_count']
-        self.stopped_epoch = state_dict['stopped_epoch']
-        self.best_score = state_dict['best_score']
-        self.patience = state_dict['patience']
+    def on_load_checkpoint(self, checkpointed_state):
+        self.wait_count = checkpointed_state['wait_count']
```

Comment:
What will happen if: 
We first train a model but our callbacks=[model_checkpoint] (no early stopping)
It saved a checkpoint. Now we slightly change the code to 
callbacks==[model_checkpoint, early_stopping] (we added ES). 
We resume training with our new code, will this break now? Trying to load the state from the checkpoint when we can't?